pkg/app/pipectl/cmd/installation/controlplane.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	input.Logger.Info(stdout.String())
+	input.Logger.Info("Installed the controlplane's components successfully.")
+
+	return nil
+}
+
+func (c *controlplane) findHelm() (string, error) {
+	fi, err := os.Stat(path.Join(c.toolsDir, helmBinaryName))
+	if err != nil && !os.IsNotExist(err) {
+		return "", err
+	}
+
+	// If the Helm executable binary exists in tools dir, use it.
+	if fi != nil {
+		return path.Join(c.toolsDir, helmBinaryName), nil
+	}
+
+	// If the Helm executable binary exists in $PATH, use it.
+	path, err := exec.LookPath("helm")
+	if err != nil {
+		return "", err
+	}
+	return path, nil
+}
+
+func (c *controlplane) buildHelmArgs() ([]string, error) {
+	args := []string{
+		"upgrade",
+		"--install",
+		helmReleaseName,
+		helmChartRepoName,
+		"--version",
+		c.version,
+		"--namespace",
+		c.namespace,
+		"--create-namespace",
+	}
+
+	if c.quickstart {
+		args = append(args,
+			"--values",
+			fmt.Sprintf(helmQuickstartValueRemotePath, c.version),
+		)
+		return args, nil
+	}
+
+	if c.configFile == "" {
+		return nil, fmt.Errorf("missing required fields: config-file")
+	}
+
+	args = append(args,
+		"--set-file",
+		fmt.Sprintf("config.data=%s", c.configFile),
+	)
+
+	if c.encryptionKeyFile != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.encryptionKey.data=%s", c.encryptionKeyFile),
+		)
+	} else {
+		// Generate encryption key data.
+		encryptionKeyFile, err := generateEncryptionKey()
+		if err != nil {
+			return nil, err
+		}
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.encryptionKey.data=%s", encryptionKeyFile),
+		)
+	}
+
+	if c.firestoreServiceAccount != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.firestoreServiceAccount.data=%s", c.firestoreServiceAccount),
+		)
+	}
+
+	if c.cloudSQLServiceAccount != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.cloudSQLServiceAccount.data=%s", c.cloudSQLServiceAccount),
+		)
+	}
+
+	if c.gcsServiceAccount != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.gcsServiceAccount.data=%s", c.gcsServiceAccount),
+		)
+	}
+
+	if c.minioAccessKey != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.minioAccessKey.data=%s", c.minioAccessKey),
+		)
+	}
+
+	if c.minioSecretKey != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.minioSecretKey.data=%s", c.minioSecretKey),
+		)
+	}
+
+	if c.internalTLSKey != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.internalTLSKey.data=%s", c.internalTLSKey),
+		)
+	}
+
+	if c.internalTLSCert != "" {
+		args = append(args,
+			"--set-file",
+			fmt.Sprintf("secret.internalTLSCert.data=%s", c.internalTLSCert),
+		)
+	}
+
+	if c.values != "" {
+		args = append(args, "--values", c.values)
+	}
+
+	return args, nil
+}
+
+func generateEncryptionKey() (string, error) {
+	script := fmt.Sprintf(generateEncryptionKeyScriptTmpl, defaultEncryptionKeyFile)
```

Comment:
It would not good if this command overrides any user's file, so how about checking the existence of that name and using an additional suffix if needed?

Reply:
That's true 👍 addressed by https://github.com/pipe-cd/pipecd/pull/3835/commits/27465fb1168baf82e6b8f6a2d6252ad6d7d9f187 🙏